src/rulesets/Base/distributed.jl

```diff
@@ -0,0 +1,61 @@
+#####
+##### `pmap`
+#####
+
+# Now that there is a backwards rule for zip (albeit only in Zygote),
+# it should be fine to deal with only a single collection X
+function rrule(config::RuleConfig{>:HasReverseMode}, ::typeof(pmap), f, p::AbstractWorkerPool, X; kwargs...)
+    project_X = ProjectTo(X)
+
+    darr = dfill([], (nworkers(p) + 1,), vcat(myid(), workers(p))) # Include own proc to handle empty worker pool
+
+    function forw(x)
+        y, back = rrule_via_ad(config, f, x)
+        push!(darr[:L][1], back)
+        return y, myid(), length(darr[:L][1])
+    end
+
+    ys_IDs_indices = pmap(forw, p, X; kwargs...)
+    ys = getindex.(ys_IDs_indices, 1) # the primal values
+    IDs = getindex.(ys_IDs_indices, 2) # remember which processors handled which elements of X
+    indices = getindex.(ys_IDs_indices, 3) # remember the index of the pullback in the array on each processor
+    output_sz = size(ys)
+ 
+    # create a list of positions in X handled by each processor
+    unique_IDs = sort(unique(IDs))
```

Comment:
What's gained by sorting these?

And why not `sort!(unique!(`?

Reply:
I still need to keep the original IDs around, but I can get rid of a mutation.

I sorted so that I could match each ID to the unique processor quickly with `searchsortedfirst`
![IMG_0325](https://user-images.githubusercontent.com/43899491/159338554-371bb1c7-3120-4438-a776-2ffc6b4a755d.jpg)
.